src/main/kotlin/com/meltwater/docker/compose/ProcessWrapper.kt

```diff
@@ -0,0 +1,68 @@
+package com.meltwater.docker.compose
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.io.BufferedReader
+import java.io.InputStream
+import java.util.concurrent.TimeUnit
+import kotlin.concurrent.thread
+
+class ProcessWrapper(val pb: ProcessBuilder, stdOutListener: (String) -> Unit, stdErrListener: (String) -> Unit){
+
+    val LOGGER: Logger = LoggerFactory.getLogger(ExecUtils::class.java)
+
+    val process: Process
+    val stdOutReader: Thread
+    val stdErrReader: Thread
+    var exitCode: Int = 0
```

Comment:
If you ask for the exit code before termination you get 0 - a bit dangerous. Change to -1